src/Symfony/Bundle/FrameworkBundle/Resources/config/services.xml

```diff
@@ -120,5 +120,11 @@
             <argument type="service" id="request_stack" />
             <tag name="kernel.event_subscriber" />
         </service>
+
+        <service id="slugger" class="Symfony\Component\String\Slugger\AsciiSlugger">
```

Comment:
should be in a file loaded only when the component class exists.

Reply:
Not necessarily: my aim here is to provide a nicer error message when the String component is required. This means replacing this definition with the stdClass definition when the component is not present, asking to install the component. This also means that this definition can be stored here IMO.